tinydir.h

```diff
@@ -704,6 +704,17 @@ if (errno)
 	base_name =basename(file_name_buf);
 #endif
 
+	/* Special case: if the path is a root dir, open the parent dir as the file */
+	if (_tinydir_strlen(base_name) == 0)
+	{
+		memset(file, 0, sizeof * file);
+		file->is_dir = 1;
+		file->is_reg = 0;
+		_tinydir_strcpy(file->path, dir_name);
+		file->extension = file->path + _tinydir_strlen(file->path);
```

Comment:
I think this will work, but since in other cases file->extension points into the file->name array, for consistency this could be `file->extension = file->name`.